jobserver/migrations/0027_add_membership_org_and_project.py

```diff
@@ -0,0 +1,106 @@
+                        primary_key=True,
+                        serialize=False,
+                        verbose_name="ID",
+                    ),
+                ),
+                ("roles", jobserver.fields.RolesField(default=list)),
+                (
+                    "created_by",
+                    models.ForeignKey(
+                        null=True,
+                        on_delete=django.db.models.deletion.SET_NULL,
+                        related_name="org_invitations",
+                        to=settings.AUTH_USER_MODEL,
+                    ),
+                ),
+                (
+                    "org",
+                    models.ForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE,
+                        related_name="members",
+                        to="jobserver.org",
+                    ),
+                ),
+                (
+                    "user",
+                    models.ForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE,
+                        related_name="org_memberships",
+                        to=settings.AUTH_USER_MODEL,
+                    ),
+                ),
+            ],
+            options={
+                "unique_together": {("org", "user")},
+            },
+        ),
+        migrations.CreateModel(
+            name="ProjectMembership",
+            fields=[
+                (
+                    "id",
+                    models.AutoField(
+                        auto_created=True,
+                        primary_key=True,
+                        serialize=False,
+                        verbose_name="ID",
+                    ),
+                ),
+                ("roles", jobserver.fields.RolesField(default=list)),
+                (
+                    "created_by",
+                    models.ForeignKey(
+                        null=True,
+                        on_delete=django.db.models.deletion.SET_NULL,
+                        related_name="project_invitations",
+                        to=settings.AUTH_USER_MODEL,
+                    ),
+                ),
+                (
+                    "project",
+                    models.ForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE,
+                        related_name="members",
+                        to="jobserver.project",
+                    ),
+                ),
+                (
+                    "user",
+                    models.ForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE,
+                        related_name="project_memberships",
+                        to=settings.AUTH_USER_MODEL,
+                    ),
+                ),
+            ],
+            options={
+                "unique_together": {("project", "user")},
+            },
+        ),
+        migrations.DeleteModel(
+            name="Membership",
+        ),
```

Comment:
Is there no need to map this old data into the new model? Or am I missing the place where that happens?

Reply:
This table is empty!  There's some backstory to that…

When I added Orgs and Projects in #394 I realised we'd need a Membership table so added it there.  As part of the Onboarding work #394 was part of we realised Roles&Permissions needed to come first (hence [OFP#13](https://github.com/opensafely-core/roadmap/discussions/13)).  Then, when I was desiging that framework I realised Membership was a good place to hold Local Roles for a Project, but also an Org.  Since the old Membership table was empty I didn't bother trying to refactor it via migrations.

Reply:
Well, if it's empty... :-)

Reply:
@benbc, context you may be missing (apols if not) is that these migrations are code generated and consumed by the Django ORM and are generally considered a black box.